client-cli/src/command/wallet_command.rs

```diff
@@ -122,6 +171,135 @@ impl WalletCommand {
         Ok(())
     }
 
+    fn export<T: WalletClient>(
+        wallet_client: T,
+        name: &Option<String>,
+        from_file: &Option<PathBuf>,
+        to_file: &Option<PathBuf>,
+    ) -> Result<()> {
+        let mut wallet_info_list = vec![];
+        let mut error_wallets = vec![];
+        match (name, from_file) {
+            (Some(_name), Some(_from_file)) => {
+                return Err(Error::new(
+                    ErrorKind::InvalidInput,
+                    "only need name or from_file",
+                ))
+            }
+            (None, None) => {
+                return Err(Error::new(
+                    ErrorKind::InvalidInput,
+                    "need name or from_file",
+                ))
+            }
+            (Some(names), None) => {
+                for name in names.split(',') {
+                    let enckey =
+                        ask_seckey(Some(&format!("Enter authentication token for {}: ", name)))?;
+                    let wallet_info = wallet_client.export_wallet(name, &enckey)?;
+                    wallet_info_list.push(wallet_info);
+                }
+            }
+            (None, Some(from_file)) => {
+                let mut file =
+                    File::open(from_file).chain(|| (ErrorKind::IoError, "Unable to open file"))?;
+                let mut settings: String = String::new();
+                file.read_to_string(&mut settings)
+                    .chain(|| (ErrorKind::IoError, "Unable to read from file"))?;
+                let wallet_settings: Value = serde_json::from_str(&settings)
+                    .chain(|| (ErrorKind::InvalidInput, "Invalid wallet info"))?;
+                let wallet_settings = wallet_settings.as_array().chain(|| {
+                    (
+                        ErrorKind::InvalidInput,
+                        "from_file should contains a list of setting",
+                    )
+                })?;
+                for setting in wallet_settings {
+                    let name = setting["name"]
+                        .as_str()
+                        .chain(|| (ErrorKind::InvalidInput, "expect name"))?;
+                    let key = setting["auth_token"]
+                        .as_str()
+                        .chain(|| (ErrorKind::InvalidInput, "Expect auth_token"))?;
+                    let enckey = parse_hex_enckey(&key).map_err(|_e| {
+                        error(&format!("error auth_token of wallet {}", name));
+                    });
```

Comment:
probably giving it a data type "Vec<WalletExportSettings>" with serde implementations would be less error-prone ?

Reply:
Ok